### PR TITLE
Add reverse proxy setting

### PR DESCRIPTION
**Pull request by martin-css**
## Description

Adds a new `ReverseProxy` option that controls whether the application accepts HTTP headers that can be set by a reverse proxy to indicate the real user etc. 

Currently, this only affects whether the logger respects the `X-Real-Ip` header. If set, this will be used if present. If not set, this will be ignored, even with a value as a malicious user could be trying to log a false IP address.

In the future this could also control other common headers, e.g. `X-Forwarded-For` etc. 

Big discussion point is that this is a **BREAKING CHANGE** for users currently behind a reverse proxy. The default value has been deliberately set to `false` as a user who runs this with a minimal configuration (without fully reading docs) should be left with a secure configuration by default. If being hosted behind a reverse proxy then this flag needs to be enabled. 

We should consider:

1.  Best way to communicate clear notice to existing users about breaking change
2. Should we temporarily leave default as `true` to allow for a period of bedding in then change to `false` after a suitable time? If so, how long?

## Motivation and Context

Corrects the security issue by always accepting the `X-Real-Ip` header (described in #321).

## How Has This Been Tested?

Quick and dirty test with curl - should be checked by others.

## Checklist:

- [X] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [ ] I have created a feature (non-master) branch for my PR.

Documentation has been updated but left the changelog until discussion has been had around best way to implement (if at all).

### CODE REVIEW

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -251,10 +253,12 @@ func (l *Logger) GetFileLineString(calldepth int) string {
 }
 
 // GetClient parses an HTTP request for the client/remote IP address.
-func GetClient(req *http.Request) string {
-	client := req.Header.Get("X-Real-IP")
-	if client == "" {
-		client = req.RemoteAddr
+func GetClient(reverseProxy bool, req *http.Request) string {
```
Comment (syscll) [inline pkg/logger/logger.go]: I think it's "best practice" to have `req` as the first arg when a func deals with a HTTP request.
```suggestion
func GetClient(req *http.Request, reverseProxy bool) string {
```

File: pkg/logger/logger.go [hunk 2]
```diff
@@ -308,6 +312,13 @@ func (l *Logger) SetReqEnabled(e bool) {
 	l.reqEnabled = e
 }
 
+// SetReverseProxy controls whether logging will trust headers that can be set by a reverse proxy.
+func (l *Logger) SetReverseProxy(e bool) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
```
Comment (syscll) [inline pkg/logger/logger.go]: We could remove a call to defer here and explicitly unlock the mutex at the end of the function.

Comment (martin-css) [reply pkg/logger/logger.go]: I'd used a deferred unlock as that is the standard that is followed in all other `Set*` functions in the logger.

`defer` has a small performance penalty and shouldn't be used in a very hot code path but for this initialisation code I don't think it's relevant to worry about - compared to the benefit of ensuring unlock is always called even if the function logic becomes more complicated in the future.

I'm not strongly opinionated on this either way but think the package should be consistent for all methods whether explicit calls or use of deferred unlocks is decided.

**Comment by martin-css** [comment]
The CHANGELOG has been updated and minor change requested was added. Anything else needing done for this PR?

**Comment by syscll** [comment]
@JoelSpeed @steakunderscore can you also review this please.

**Pull request merged**